### PR TITLE
storage: save created index and modified index

### PR DESCRIPTION
**Pull request by xiang90**
Save created and modified index. Made some necessary changes to the index system.

### CODE REVIEW

File: storage/key_index.go [hunk 1]
```diff
@@ -215,8 +220,9 @@ func (ki *keyIndex) String() string {
 }
 
 type generation struct {
-	ver  int64
-	revs []reversion
+	ver     int64
+	created reversion // when the generation is created (put in first reversion).
```
Comment (yichengq) [inline storage/key_index.go]: is this var needed explicitly? It is always the first element in `revs`.

**Comment by yichengq** [comment]
lgtm

File: storage/key_index.go (hunk 1 above)
Comment (xiang90) [inline storage/key_index.go]: i do not understand. the version of first one in the generation is 1. the second one is 2... when you compact it, you need to have a variable to remember the version.

Comment (yichengq) [inline storage/key_index.go]: you are right. the compact case needs the created var. lgtm!

**Pull request merged**